numpy/_core/src/common/npy_cpu_features.c

```diff
@@ -504,58 +528,97 @@ npy__cpu_init_features(void)
+                                            npy__cpu_have[NPY_CPU_FEATURE_MOVBE];
+
+
+    npy__cpu_have[NPY_CPU_FEATURE_X86_V4] = npy__cpu_have[NPY_CPU_FEATURE_X86_V3] &&
+                                            npy__cpu_have[NPY_CPU_FEATURE_AVX512F] &&
+                                            npy__cpu_have[NPY_CPU_FEATURE_AVX512CD] &&
+                                            npy__cpu_have[NPY_CPU_FEATURE_AVX512BW] &&
+                                            npy__cpu_have[NPY_CPU_FEATURE_AVX512DQ] &&
+                                            npy__cpu_have[NPY_CPU_FEATURE_AVX512VL];
+
+
+    npy__cpu_have[NPY_CPU_FEATURE_AVX512_ICL] = npy__cpu_have[NPY_CPU_FEATURE_X86_V4] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512VNNI] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512IFMA] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512VBMI] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512VBMI2] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512BITALG] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512VPOPCNTDQ] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_GFNI] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_VAES] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_VPCLMULQDQ];
+
+    npy__cpu_have[NPY_CPU_FEATURE_AVX512_SPR] = npy__cpu_have[NPY_CPU_FEATURE_AVX512_ICL] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512FP16];
+
+
+    // Legacy groups
+    npy__cpu_have[NPY_CPU_FEATURE_AVX512_KNL] = npy__cpu_have[NPY_CPU_FEATURE_AVX512F] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512CD] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512ER] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512PF];
+
+    npy__cpu_have[NPY_CPU_FEATURE_AVX512_KNM] = npy__cpu_have[NPY_CPU_FEATURE_AVX512_KNL] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX5124FMAPS] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX5124VNNIW] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512VPOPCNTDQ];
+
+    npy__cpu_have[NPY_CPU_FEATURE_AVX512_CLX] = npy__cpu_have[NPY_CPU_FEATURE_X86_V4] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512VNNI];
+
+    npy__cpu_have[NPY_CPU_FEATURE_AVX512_CNL] = npy__cpu_have[NPY_CPU_FEATURE_X86_V4] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512IFMA] &&
+                                                npy__cpu_have[NPY_CPU_FEATURE_AVX512VBMI];
```

Comment:
```suggestion
```
Not sure if we should drop legacy groups. I kept them in case there are projects that depend on them via dict `__cpu_features__`

Reply:
sounds a good call to me